### PR TITLE
Add amazon a9 header bidding

### PR DESCRIPTION
**Pull request by dwick**
ticket: https://reddit.atlassian.net/browse/ADS-1909

Currently this doesn't have support for some params that r2 uses (`subreddit`, `SIS_category`, etc.), but the docs amazon passed me seem incomplete and I'm not sure how we're supposed to do that with a SPA. r2 passes them via `apstag.init`, which can only be called once. I'll reach out and fix that up if possible. This should still work however.

### CODE REVIEW

File: src/lib/dfp.js [hunk 1]
```diff
@@ -78,4 +93,18 @@ export const setupGoogleTag = () => {
...
+  // eslint-disable-next-line
+  !function(a9,a,p,s,t,A,g){if(a[a9])return;function q(c,r){a[a9]._Q.push([c,r])}a[a9]={init:function(){q('i',arguments)},fetchBids:function(){q('f',arguments)},_Q:[]};A=p.createElement(s);A.async=!0;A.src=t;g=p.getElementsByTagName(s)[0];g.parentNode.insertBefore(A,g)}('apstag',window,document,'script','//c.amazon-adsystem.com/aax2/apstag.js');
+
+  // this is defined in the above IIF
+  window.apstag.init({
+    pubID: '3379',
+    adServer: 'googletag',
+    bidTimeout: 2e3,
+    params: {
+      position: 'feed',
```
Comment (griffinmichl) [inline src/lib/dfp.js]: What's this? Are we not setting up header bidding for banners as well? Banners are worth more

File: src/lib/dfp.js [hunk 2]
```diff
@@ -40,10 +40,25 @@ export const defineSlot = (element, {
...
+        slots: [{
+          slotID: id,
+          sizes,
+        }],
+        timeout: 2e3,
+      }, (bids) => {
+        // we only request a single bid.
+        const bid = bids[0];
+
+        ['amznbid', 'amzniid'].forEach(function(key) {
```
Comment (griffinmichl) [inline src/lib/dfp.js]: Has mike setup DFP for mweb ads to handle these keys?

File: src/lib/dfp.js (hunk 1 above)
Comment (dwick) [inline src/lib/dfp.js]: i forgot we had banners

File: src/lib/dfp.js (hunk 2 above)
Comment (dwick) [inline src/lib/dfp.js]: amazon does this i think

File: src/lib/dfp.js (hunk 1 above)
Comment (griffinmichl) [inline src/lib/dfp.js]: We'll probably want to get setup for those as well. They're at least 5x the value of in feed.

File: src/lib/dfp.js (hunk 2 above)
Comment (griffinmichl) [inline src/lib/dfp.js]: Are you sure? These are keys being passed to DFP. How would amazon configure them?

**Comment by griffinmichl** [comment]
🐟 Once this is tested manually

**Comment by dwick** [comment]
screen shot 2017-08-15 at 4 45 12 pm

💇

**Comment by griffinmichl** [comment]
🐟

**Pull request merged**